templates/js_fcl_main.tmpl

```diff
@@ -0,0 +1,39 @@
+/**
+    This binding file was auto generated based on FLIX template v{{.Version}}. 
+    Changes to this file might get overwritten 
+**/
+
+import * as fcl from "@onflow/fcl"
+{{- if .IsLocalTemplate }}
+import flixTemplate from "{{.Location}}"
+{{- else}}
+const flixTemplate = "{{.Location}}"
+{{- end}}
+
+// TODO: add runtime check for parameters consistency
+const parameterNames = [
+  {{- range $index, $ele := .Parameters -}}
+     {{if $index}}, {{end}}"{{$ele.Name}}"
+  {{- end -}}
+];
+ {{"\n"}}/**
+* {{.Description}}{{"\n"}}
+   {{- if gt (len .Parameters) 0 -}}
+* @param {Object} Parameters - parameters for the cadence
+    {{- range $index, $ele := .Parameters -}}
+{{"\n"}}* @param {{"{"}} {{$ele.JsType}} {{"}"}} Parameters.{{$ele.Name}} - {{$ele.Description}}: {{$ele.CadType}}
+    {{- end -}}
+ {{ end -}}
+ {{- if not .IsScript -}}
+{{"\n"}}* @returns {Promise<string>} - returns a promise which resolves to the transaction id
+ {{- end -}}
+{{"\n"}}*/
+
+{{if .IsScript}}
+{{- template "js_fcl_script.tmpl" .}}
```

Comment:
How does this differentiate between different js libs?

Reply:
It doesn't it's fcl specific. Not sure what the pattern would be if fcl wouldn't be used.